go/oasis-test-runner/scenario/e2e/runtime/history_reindex.go

```diff
@@ -82,14 +85,116 @@ func (sc *historyReindexImpl) Clone() scenario.Scenario {
 }
 
 func (sc *historyReindexImpl) Run(ctx context.Context, childEnv *env.Env) error {
-	cli := cli.New(childEnv, sc.Net, sc.Logger)
-
 	// Start the network.
 	if err := sc.Net.Start(); err != nil {
 		return err
 	}
 
 	// Wait for enough block to ensure pruning on the compute node.
+	if err := sc.ensureComputePruning(ctx); err != nil {
+		return fmt.Errorf("failed to ensure compute node pruning: %w", err)
+	}
+
+	// Restart compute worker with configured runtime.
+	if err := sc.restartCompute(ctx, childEnv); err != nil {
+		return fmt.Errorf("failed to restart compute node: %w", err)
+	}
+
+	// Run client to ensure runtime works.
+	sc.Logger.Info("Starting the basic client")
+	if err := sc.RunTestClientAndCheckLogs(ctx, childEnv); err != nil {
+		return fmt.Errorf("failed to run test client and check logs: %w", err)
+	}
+
+	lastRound, err := sc.fetchLatestRound(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to fetch default client latest round: %w", err)
+	}
+
+	// Start new clients, all with different history indexer batch size configuration.
+	// For every client, ensure reindex works by comparing all blocks of default client.
+	for _, batchSize := range []uint16{1, 2, 3, 1000} {
+		if err := sc.startNewClientAndTestReindex(ctx, lastRound, batchSize); err != nil {
+			return fmt.Errorf("wrong history reindex (batchSize: %d): %w", batchSize, err)
+		}
+	}
+
+	return nil
+}
+
+func (sc *historyReindexImpl) startNewClientAndTestReindex(
```

Comment:
I would expect this method to be lower, as it is used at the end of the `run`. So if `run` calls methods `A`, `B`, `C` in that order, I would also expect that method `B` is defined below `A` and `C` below `B`. Unless these are publlic methods, which should usually be always on the top.